backend/src/main/java/me/performancereservation/domain/reservation/service/redis/RedisSeatReservationService.java

```diff
@@ -0,0 +1,117 @@
+import me.performancereservation.domain.reservation.service.SeatReservationService;
+import me.performancereservation.global.exception.ErrorCode;
+import me.performancereservation.global.storage.redis.RedisReservationService;
+import me.performancereservation.global.storage.redis.RedisSeatService;
+import org.springframework.stereotype.Service;
+
+@Service("redisSeatReservationService")
+@RequiredArgsConstructor
+public class RedisSeatReservationService implements SeatReservationService {
+    private final ReservationRepository reservationRepository;
+    private final PerformanceScheduleRepository performanceScheduleRepository;
+
+    private final RedisSeatService redisSeatService;
+    private final RedisReservationService redisReservationService;
+
+    private final ReservationMapper reservationMapper;
+
+    /**
+     * 공연 회차에 대해 좌석 선점 후 예약 정보를 생성하고 나서
+     * 결제 만료 처리를 위한 시간 정보 등록까지 수행함
+     *
+     * 1. Redis에서 좌석 선점 (동시성 제어)
+     * 2. 좌석이 부족하면 롤백 + 예외 처리
+     * 3. Reservation 저장 (상태: PAYMENTS_PENDING - 결제 대기)
+     * 4. Redis ZSet에 예약(결제) 만료 정보 등록
+     *
+     * TODO 추후 어드민 무통장 결제 승인 메서드에서 티켓 수량만큼을 해당 Schedule에 차감 반영해야함
+     *
+     * @param scheduleId 공연 회차 ID
+     * @param userId 예약하는 유저 ID
+     * @param quantity 예약한 티켓 수
+     * @return ReservationResponse
+     */
+    @Transactional
+    @Override
+    public ReservationResponse reserve(Long scheduleId, Long userId, int quantity) {
+        // Redis에서 좌석 1개 먼저 차감 (RDB 접근 없이 선 예약 선점 - 빠른 필터링이 가능한게 장점)
+        redisSeatService.safeDecrement(scheduleId, quantity); // 내부에 Redis 좌석 선점 관련 예외처리 있음
+
+        // 예약 엔티티 저장 (status -> PAYMENTS_PENDING (결제 대기))
+        Reservation reservation = reservationRepository.save(
+                Reservation.builder()
+                        .scheduleId(scheduleId)
+                        .userId(userId)
+                        .quantity(quantity)
+                        .status(ReservationStatus.PAYMENTS_PENDING)
+                        .build()
+        );
+
+        // 결제 대기 중인 예약에 대한 결제 만료 시간을 Redis ZSet에 등록
+        // ReservationExpirationScheduler가 주기적으로 조회해서 만료 시간 초과 시 예약을 자동 취소
+        redisReservationService.addToPendingExpirationQueue(reservation.getId());
+
+        // response dto mapping용도로 schedulePerformanceInfo 데이터 모델 조회
+        SchedulePerformanceInfo schedulePerformanceInfo = performanceScheduleRepository.findSchedulePerformanceInfoByScheduleId(scheduleId)
+                .orElseThrow(() -> ErrorCode.SCHEDULE_NOT_FOUND.domainException("해당 id의 공연 회차 없음 : " + scheduleId));
+
+        return reservationMapper.toResponseDto(reservation, schedulePerformanceInfo);
+    }
+
+
+    /**
+     * 유저가 직접 본인의 예약을 취소함
+     * 만약 결제완료가 된 상태면 환불객체를 생성하고 취소 대기상태가 돼야함
+     * 결제완료가 안된 상태면 그대로 취소완료처리
+     *
+     * @param reservationId 예약 ID
+     * @param userId 예약한 유저 ID
+     */
+    @Transactional
+    @Override
+    public void cancel(Long reservationId, Long userId) {
```

Comment:
수혁님 저는 개인적으로 긴 내용이 밑으로 쌓이는 형태를 좋아합니다.
```suggestion
    @Override
    @Transactional
    public void cancel(Long reservationId, Long userId) {
```

아 절대 불편하다는 건 아닙니다. 전혀 신경 안 쓰셔도 됩니다. 
그냥 그렇다고요....

장난입니다 ㅋㅋ

Reply:
ㅋㅋㅋㅋㅋ 수정할게요